tests/utils/factories.py

```diff
@@ -17,26 +20,126 @@
 from citrine.resources.material_spec import MaterialSpec
 from citrine.resources.material_template import MaterialTemplate
 from citrine.resources.process_template import ProcessTemplate
-from gemd.entity.link_by_uid import LinkByUID
+from citrine.resources.table_config import TableConfigInitiator
+
+from gemd import LinkByUID, EmpiricalFormula, FileLink
+from gemd.enumeration import SampleType
+
+
+class AugmentedProvider(Provider):
+    def random_formula(self, count: int = None, elements: Set[str] = None) -> str:
+        """Generate a random, well-formed chemical formula.  Likely non-physical."""
+        if elements is None or len(elements) == 0:
+            elements = list(EmpiricalFormula.all_elements())  # Must be Sequence
+        if count is None:
+            count = self.generator.random.randrange(1, 5)
+        components = sorted(self.generator.random.sample(elements, count))
+        # Use weights to bias toward looking more real-ish
+        amounts = self.generator.random.choices([1, 2, 3, 4, 5], weights=[40, 40, 10, 10, 2], k=count)
+        return "".join(f"({c}){a}" for c, a in zip(components, amounts))
+
+    def random_smiles(self) -> str:
+        """Generate a random, well-formed chemical formula.  Likely non-physical."""
+        element_weights = {  # With selection weights
+            "C": 100,
+            "O": 20,
+            "N": 20,
+            "P": 2,
+            "S": 2,
+            "B": 2,
+            "F": 1,
+            "Cl": 1,
+            "Br": 1,
+            "I": 1
+        }
+        valence = {
+            "B": 3,
+            "C": 4,
+            "N": 3,
+            "O": 2,
+            "P": 3,
+            "S": 4,
+            "F": 1,
+            "Cl": 1,
+            "Br": 1,
+            "I": 1
+        }
+        bonds = ['', '=', '#', '$']
+
+        elements = [x for x in element_weights]
+        weights = [element_weights[x] for x in element_weights]
```

Comment:
`elements, weights = element_weights.items()`

Reply:
I think that's backwards:
`ValueError: too many values to unpack (expected 2)`

Reply:
...ignore me, it's early. XD

Reply:
You could just do `element_weights.keys()` and `element_weights.values()`, although I'm not too adamant about that.